PicPi/AddFrameViewController.swift

```diff
@@ -13,6 +13,8 @@ class AddFrameViewController: UIViewController, UITextFieldDelegate  {
   let ipTextField = customUITextFieldWithDecimalPad()
   var margins: UILayoutGuide!
   let mqttManager =  MQTTManager()
+  private let userPreferences = injectUserPreferences()
```

Comment:
Property Injection ?

Reply:
Something like it lol so view controllers don't get tested like business logic types so injecting it here vs through the init is okay

If the inject method needs arguments (eg `injectFrameAPI(host: myPIURL)`) then the service might need to be injected from the init